File: Data Structures & Algorithms/anagram-groups/submission-10.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, ArrayList> res = new HashMap<>();
        for(String s : strs) {
            int[] count = new int[26];
            for(char c : s.toCharArray()) {
                count[c - 'a']++;
            }
            String cnt = Arrays.toString(count);
            res.putIfAbsent(cnt, new ArrayList<String>());
            res.get(cnt).add(s);
        }

        return new ArrayList(res.values());
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-10.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if not s:
            return 0
        if len(s) < 2:
            return 1

        l, r = 0, 1
        res = 1
        ss = s[l]

        while r < len(s):
            if s[r] not in ss:
                ss += s[r]
                res = max(res, len(ss))
                r += 1
            else:
                l += 1
                ss = s[l]
                r = l+1
        print(ss)
        return res